### PR TITLE
EIP-1102: Clear approvals on rejection

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request updates the implementation of EIP-1102 in two ways:

1. If a user rejects approval on a given origin, we always clear cached approvals for that origin.
2. Dynamically-injected script tags are cleaned up after use.

Refs #5683 
Refs #5688

### CODE REVIEW

**Review by whymarrh — CHANGES_REQUESTED** [review]
Can we drop the `package-lock.json` changes from this PR?

File: app/scripts/contentscript.js [hunk 1]
```diff
@@ -37,8 +37,10 @@ function injectScript (content) {
   try {
     const container = document.head || document.documentElement
     const scriptTag = document.createElement('script')
+    scriptTag.setAttribute('async', false)
     scriptTag.textContent = content
     container.insertBefore(scriptTag, container.children[0])
+    container.removeChild(scriptTag)
```
Comment (whymarrh) [inline app/scripts/contentscript.js]: Should we remove this after the script has loaded?

**Comment by bitpshr** [comment]
@whymarrh good catch, I've been changing Node versions locally for unrelated reasons. Lockfile reverted.

File: app/scripts/contentscript.js (hunk 1 above)
Comment (bitpshr) [reply app/scripts/contentscript.js]: @whymarrh it was my understanding that dynamically-added script tags with `async=false` will be executed synchronously. I verified this crudely using the following code in the console:

```js
const container = document.head || document.documentElement
const scriptTag = document.createElement('script')
scriptTag.setAttribute('async', false)
scriptTag.textContent = 'for(var i = 0; i< 10000; i++) { console.log(i); }'
container.insertBefore(scriptTag, container.children[0])
console.log('done')
```

Comment (whymarrh) [reply app/scripts/contentscript.js]: Yup, that makes sense (script tags are synchronous, I forgot that)

**Review by whymarrh — DISMISSED** [review]

**Review by brunobar79 — APPROVED** [review]

**Pull request merged**